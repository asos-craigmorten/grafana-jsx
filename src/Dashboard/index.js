/**
 * REF: https://grafana.com/docs/grafana/latest/reference/dashboard/#json-fields
 */

import PropTypes from "prop-types";
import Annotation from "../Annotation";
import Annotations from "../Annotations";
import Links from "../Links";
import Panels from "../Panels";
import Templates from "../Templates";
import Time from "../Time";
import TimePicker from "../TimePicker";

const Dashboard = ({
  children,
  editable = true,
  hideControls = false,
  graphTooltip = 0,
  refresh,
  style = "dark",
  tags = [],
  title,
  timezone = "browser",
  uid,
  ...props
}) => (
  <dashboard
    editable={editable}
    graphTooltip={graphTooltip}
    hideControls={hideControls}
    id={null}
    refresh={refresh}
    style={style}
    tags={tags}
    timezone={timezone}
    title={title}
    uid={uid}
    {...props}
  >
    <Annotations>
      <Annotation />
    </Annotations>
    <Links />
    <Panels />
    <Templates />
    <Time />
    <TimePicker />
    {children}
  </dashboard>
);

Dashboard.PropTypes = {
  children: PropTypes.any,
  editable: PropTypes.bool,
  hideControls: PropTypes.bool,
  graphTooltip: PropTypes.oneOf([0, 1, 2]),
  refresh: PropTypes.string,
  style: PropTypes.oneOf(["dark", "light"]),
  tags: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string.isRequired,
  timezone: PropTypes.oneOf(["browser", "utc"]),
  uid: PropTypes.string,
};

export default Dashboard;
