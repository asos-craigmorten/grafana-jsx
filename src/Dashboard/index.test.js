import Dashboard from "./";

jest.mock("../Annotation", () => () => <test-annotation />);
jest.mock("../Annotations", () => ({ children }) => (
  <test-annotations>{children}</test-annotations>
));
jest.mock("../Links", () => () => <test-links />);
jest.mock("../Panels", () => () => <test-panels />);
jest.mock("../Templates", () => () => <test-templates />);
jest.mock("../Time", () => () => <test-time />);
jest.mock("../TimePicker", () => () => <test-timepicker />);

const requiredProps = {
  title: "test-title",
};

const withDefaultsProps = {
  ...requiredProps,
  refresh: "test-refresh",
  style: "dark",
  timezone: "browser",
  uid: "test-uid",
};

const withAllProps = {
  ...withDefaultsProps,
  editable: false,
  hideControls: true,
  graphTooltip: 2,
  tags: ["test-tag-1", "test-tag-2"],
};

describe("Dashboard", () => {
  describe("when passed required props", () => {
    it("should render", () => {
      expect(<Dashboard {...requiredProps} />).toMatchSnapshot();
    });
  });

  describe("when passed required props and props which don't have defaults", () => {
    it("should render", () => {
      expect(<Dashboard {...withDefaultsProps} />).toMatchSnapshot();
    });
  });

  describe("when passed all props", () => {
    it("should render", () => {
      expect(<Dashboard {...withAllProps} />).toMatchSnapshot();
    });
  });

  describe("when passed children", () => {
    it("should render", () => {
      expect(
        <Dashboard {...requiredProps}>
          <test-child test-child-prop={Symbol("test-child-value")} />
        </Dashboard>
      ).toMatchSnapshot();
    });
  });
});
