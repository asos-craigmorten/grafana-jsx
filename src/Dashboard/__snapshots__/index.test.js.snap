// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard when passed all props should render 1`] = `
Object {
  "editable": false,
  "graphTooltip": 2,
  "hideControls": true,
  "id": null,
  "refresh": "test-refresh",
  "style": "dark",
  "tags": Array [
    "test-tag-1",
    "test-tag-2",
  ],
  "test-annotations": Object {
    "test-annotation": Object {},
  },
  "test-links": Object {},
  "test-panels": Object {},
  "test-templates": Object {},
  "test-time": Object {},
  "test-timepicker": Object {},
  "timezone": "browser",
  "title": "test-title",
  "uid": "test-uid",
}
`;

exports[`Dashboard when passed children should render 1`] = `
Object {
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "style": "dark",
  "tags": Array [],
  "test-annotations": Object {
    "test-annotation": Object {},
  },
  "test-child": Object {
    "test-child-prop": Symbol(test-child-value),
  },
  "test-links": Object {},
  "test-panels": Object {},
  "test-templates": Object {},
  "test-time": Object {},
  "test-timepicker": Object {},
  "timezone": "browser",
  "title": "test-title",
}
`;

exports[`Dashboard when passed required props and props which don't have defaults should render 1`] = `
Object {
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "refresh": "test-refresh",
  "style": "dark",
  "tags": Array [],
  "test-annotations": Object {
    "test-annotation": Object {},
  },
  "test-links": Object {},
  "test-panels": Object {},
  "test-templates": Object {},
  "test-time": Object {},
  "test-timepicker": Object {},
  "timezone": "browser",
  "title": "test-title",
  "uid": "test-uid",
}
`;

exports[`Dashboard when passed required props should render 1`] = `
Object {
  "editable": true,
  "graphTooltip": 0,
  "hideControls": false,
  "id": null,
  "style": "dark",
  "tags": Array [],
  "test-annotations": Object {
    "test-annotation": Object {},
  },
  "test-links": Object {},
  "test-panels": Object {},
  "test-templates": Object {},
  "test-time": Object {},
  "test-timepicker": Object {},
  "timezone": "browser",
  "title": "test-title",
}
`;
